### DISCUSSION TITLE
Way to output the notification content to a local file

### DISCUSSION THREAD

**Discussion by ctippur** [cloud-custodian/cloud-custodian#7326] [General]
Hello,

I am struggling to filter contents of a "custodian run" output to contain just the resource, policy name and the violation/trigger.

I tried to create a report off of the dryrun and I see that there is a lot of data that cannot be consumed. I can write to sqs and read it via lambda but looking for a simpler solution like read from a local file.
Is there/should there be a way to output the notification content to a local file (in the same structure as ```-s /tmp/sample/ ``` option?
I can then create a concise report based on this.

Appreciate your input.

Shekar

**Comment by thisisshi** [comment]
hm, perhaps adding an additional local file `transport` type and have that push all the messages into a json file could work

```yaml
policies:
  - type: notify-local
    resource: ec2
    actions:
      - type: notify
        transport:
          type: local
          directory: /path/to/notify.json
```

**Comment by ctippur** [comment]
Hmm.. not sure I get it

```
policies:
- filters:
  - type: metrics
    name: CPUUtilization
    days: 4
    value: 10
    op: less-than
  - type: metrics
    name: NetworkIn
    days: 4
    value: 500
    op: less-than
  actions:
  - type: notify
    transport:
      type: local
      directory: /tmp/notify.json
  description: |
    Yellow: An instance had 10% or less daily average CPU utilization and 5 MB orless network I/O on at least 4 of the previous 14 days.
  name: low-utilization-amazon-ec2-instances
  resource: ec2
```
Thats my policy which results in an error:
```
Exception: Failed to validate on policy 'metrics' is not one of ['event']

Failed validating 'enum' in schema[0]['properties']['type']:
    {'enum': ['event']}

On instance['type']:
    'metrics' 
 {'filters': [{'type': 'metrics', 'name': 'CPUUtilization', 'days': 4, 'value': 10, 'op': 'less-than'}, {'type': 'metrics', 'name': 'NetworkIn', 'days': 4, 'value': 500, 'op': 'less-than'}], 'actions': [{'type': 'notify', 'transport': {'type': 'local', 'directory': '/tmp/notify.json'}}], 'description': 'Yellow: An instance had 10% or less daily average CPU utilization and 5 MB orless network I/O on at least 4 of the previous 14 days.\n', 'name': 'low-utilization-amazon-ec2-instances', 'resource': 'ec2'} is not valid under any of the given schemas

Failed validating 'anyOf' in schema['properties']['policies']['items']:

```

**Comment by thisisshi** [comment]
Sorry, didn’t mean to imply that local is a transport that’s currently supported, meant to say that might be a pattern we could implement

**Comment by kapilt** [comment]
a couple of things, cli reporting supports adding arbitrary resource fields to its output using the --field, albeit escaping in jmespath and on the shell gets a bit obtuse for some of the annotations prefixes we use for why a given resource matched a filter. 

we also support an invoke-lambda action for one off custom integrations on sending the policy name, resource, why the resource matched a given policy to an integration lambda.
